userspace/libsinsp/container.cpp

```diff
@@ -489,72 +635,90 @@ bool sinsp_container_engine_docker::resolve(sinsp_container_manager* manager, si
 		return false;
 
 	tinfo->m_container_id = container_info.m_id;
-	if (!manager->container_exists(container_info.m_id))
+	if (!manager->container_exists(container_info.m_id) &&
+	    m_pending_requests.find(container_info.m_id) == m_pending_requests.end())
```

Comment:
I think it'd be a good idea to have some cap on the number of outstanding requests. It can be pretty high (even hundreds). I just don't want to have a degenerate case where a docker daemon failure results in an endless set of pending requests.

Reply:
I agree, will do